### DISCUSSION TITLE
Release 0.9.2

### DISCUSSION THREAD

**Discussion by TheR1D** [TheR1D/shell_gpt#268] [Announcements]
This release includes shell integrations for ZSH and Bash.
* Added simple shell integration for ZSH.
* Added simple shell integration for Bash.
* New hidden option `--install-integration` which will update user shell profile.

# Shell Integration
Shell integration allows you to use Shell-GPT in your terminal with hotkeys. It is currently available for Bash and ZSH. It will allow you to have `sgpt` completions in your shell history, and also edit suggested commands right away.

https://github.com/TheR1D/shell_gpt/assets/16740832/bead0dab-0dd9-436d-88b7-6abfb2c556c1

# Installation
```shell
pip install --upgrade shell-gpt
sgpt --install-integration
# Restart your terminal to apply changes.
```

# Usage
In your terminal enter natural language request, and press `Ctrl+l` to get shell command suggestion. It will replace your current input line (buffer) with the GPT suggested command.
```text
List all txt files in current folder (Ctrl+l)
# -> ls *.txt
```
Special thanks to @chinarjoshi 

<hr /><em>This discussion was created from the release <a href='https://github.com/TheR1D/shell_gpt/releases/tag/0.9.2'>0.9.2</a>.</em>

**Comment by dummy-work-account** [comment]
@TheR1D why did you choose to put the shell-integration scripts on a separate branch from main? is this a standard practice that I'm not aware of?

**Reply by TheR1D** [reply]
Initially, I intended to publish shell integration with several features as a "beta" release, while keep working on feature branch. However, we ended up rolling out a more simplified version of the integration. In the upcoming updates, we will be moving the shell scripts to the main branch.

**Comment by bharatagarwal** [comment]
Small suggestion:

For the demo screencast, use something with serifs to indicate the command. I thought the hotkey was `Ctrl-I` instead of `Ctrl-L`

PS: Love the tool :)

**Reply by darkside369** [reply]
thanks for this comment, i was trying to figure out why its not working and i was pressing "I"

**Comment by chinarjoshi** [comment]
Important change: I have moved the shell script to a separate repository to decouple it from the Python codebase, address the installation issue, and add complete documentation and demos. This decoupling allows the script to be used with any command-line GPT program, while keeping shell_gpt as the default program. I also made a [pr to ohmyzsh](https://github.com/ohmyzsh/ohmyzsh/pull/11779) to streamline installation.

https://github.com/chinarjoshi/englizsh/

**Comment by nixfred** [comment]
Howdy, 

Im a lower-mid level linux guy (at best) that can apt install with the best of them....

I've followed the installation instructions on a brand new Mint Linux 21.2 install and I keep getting the same error from using the command:

$ sudo docker run --rm 
--env OPENAI_API_KEY="your OPENAI API key" 
--volume gpt-cache:/tmp/shell_gpt        ghcr.io/ther1d/shell_gpt --chat rainbow "what are the colors of a rainbow"

ERROR AS FOLLOWS:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/pathlib.py", line 1116, in mkdir
    os.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: '/nonexistent/.config/shell_gpt'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/pathlib.py", line 1116, in mkdir
    os.mkdir(self, mode)
FileNotFoundError: [Errno 2] No such file or directory: '/nonexistent/.config'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/sgpt", line 5, in <module>
    from sgpt import cli
  File "/usr/local/lib/python3.11/site-packages/sgpt/__init__.py", line 1, in <module>
    from .app import main as main
  File "/usr/local/lib/python3.11/site-packages/sgpt/app.py", line 9, in <module>
    from sgpt.config import cfg
  File "/usr/local/lib/python3.11/site-packages/sgpt/config.py", line 83, in <module>
    cfg = Config(SHELL_GPT_CONFIG_PATH, **DEFAULT_CONFIG)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/sgpt/config.py", line 49, in __init__
    config_path.parent.mkdir(parents=True, exist_ok=True)
  File "/usr/local/lib/python3.11/pathlib.py", line 1120, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
  File "/usr/local/lib/python3.11/pathlib.py", line 1120, in mkdir
    self.parent.mkdir(parents=True, exist_ok=True)
  File "/usr/local/lib/python3.11/pathlib.py", line 1116, in mkdir
    os.mkdir(self, mode)
PermissionError: [Errno 13] Permission denied: '/nonexistent'

END ERROR.

Im confident that I am the problem but I cant seem to figure out what Im doing wrong....thanks for setting me straight bc I cant wait to use this tool to learn linux faster....

Phredd

**Comment by endolith** [comment]
Add support for fish plz?  🥺🙏

(Am I the only one who uses fish?)
